elife/gearman-client-php.sls

```diff
@@ -39,9 +44,51 @@ php-ppa-gearman:
             - test -e /etc/apt/sources.list.d/ondrej-ubuntu-pkg-gearman-xenial.list
         - require_in:
             - pkg: gearman-php-extension
-{% endif %}
 
 gearman-php-extension:
     pkg.installed:
         - pkgs:
             - php-gearman
+
+{% elif salt['grains.get']('osrelease') == '18.04' %}
+
+php-ppa-gearman:
+    pkg.purged:
+        - pkgs:
+            - php-gearman
+        - require_in:
+            - gearman-php-extension
+
+gearman-php-extension:
+    git.latest:
+        - name: https://github.com/wcgallego/pecl-gearman
+        - rev: gearman-2.0.5
+        - branch: master
```

Comment:
since `rev` is a tag, this could possibly be unset to leave behind a detached HEAD or similar (can't remember the details)

Reply:
yes ... I *think* in this case, because `branch` is specified as well, that revision (tag/branch) will checked out and the value for `branch` ('master') will be used locally to identify itself when you do a `git branch --show-current`. So if you're not checking out a revision from the master branch it can get misleading.